File: 0179-largest-number/0179-largest-number.py
class Solution(object):
    def cmp(self, a, b):
        if a + b > b + a:
            return -1
        return a + b < b + a
    def largestNumber(self, nums):
        a1 = list(map(str, nums))
        
        a1.sort(key=cmp_to_key(self.cmp))

        if a1[0] == '0':
            return "0"

        return ''.join(a1)

File: 0179-largest-number/0179-largest-number.java
class Solution {
    public String largestNumber(int[] nums) {
        String [] n=new String [nums.length];
        for(int i=0;i<nums.length;i++){
            n[i]=Integer.toString(nums[i]);
        }
        Arrays.sort(n,(a,b)->(b+a).compareTo(a+b));
        if(n[0].equals("0")){
            return "0";
        }
        StringBuilder res=new StringBuilder();
        for(String a:n){
            res.append(a);
        }
        System.out.println(res);
        return res.toString();
    }
}